src/cmd-cloud-prune

```diff
@@ -108,61 +108,64 @@ def main():
+
+        # For each build, iterate over arches first to minimize downloads of meta.json per arch
+        for arch in build["arches"]:
+            print(f"Processing {arch} for build {build_id}")
+            meta_prefix = os.path.join(prefix, f"{build_id}/{arch}/meta.json")
+            meta_json = get_json_from_s3(s3_client, bucket, meta_prefix)  # Download meta.json once per arch
+            images = get_supported_images(meta_json)
+            current_build = Build(id=build_id, images=images, arch=arch, meta_json=meta_json)
+
+            # Iterate over actions (policy types) to apply pruning
+            for action in ['cloud-uploads', 'images', 'build']:
+                if action not in policy[stream]:
                     continue
-                else:
-                    # OK `images` has been pruned before, but we need to check
-                    # that all the images were pruned that match the current policy.
-                    # i.e. there may be additional images we need prune
-                    previous_images_kept = previous_cleanup.get("images-kept", [])
-                    if set(images_to_keep) == set(previous_images_kept):
-                        print(f"Build {build_id} has already had {action} pruning completed")
-                        continue
-
-            for arch in build["arches"]:
-                print(f"Pruning {arch} {action} for {build_id}")
-                meta_prefix = os.path.join(prefix, f"{build_id}/{arch}/meta.json")
-                meta_json = get_json_from_s3(s3_client, bucket, meta_prefix)
-                # Make sure the meta.json doesn't contain any cloud_platform that is not supported for pruning yet.
-                images = get_supported_images(meta_json)
-                current_build = Build(id=build_id, images=images, arch=arch, meta_json=meta_json)
-
-                match action:
-                    case "cloud-uploads":
-                        prune_cloud_uploads(current_build, cloud_config, args.dry_run)
-                    # Prune through images that are not mentioned in images-keep
-                    case "images":
-                        prune_images(s3_client, current_build, images_to_keep, args.dry_run, bucket, prefix)
-                    # Fully prune releases that are very old including deleting the directory in s3 for that build.
-                    case "build":
-                        prune_build(s3_client, bucket, prefix, build_id, args.dry_run)
-                        pruned_build_ids.append(build_id)
-            # Update policy-cleanup after processing all arches for the build
-            policy_cleanup = build.setdefault("policy-cleanup", {})
+                action_duration = convert_duration_to_days(policy[stream][action])
+                ref_date = today_date - relativedelta(days=int(action_duration))
+                images_to_keep = policy.get(stream, {}).get("images-keep", [])
```

Comment:
`images_to_keep` should be the same for all runs of these loops, right? if so we can move the variable definition to the outside of these loops.